lib/Mojo/Util.pm

```diff
@@ -37,11 +38,21 @@ use constant {
 # To generate a new HTML entity table run this command
 # perl examples/entities.pl
 my %ENTITIES;
-for my $line (split "\n", join('', <DATA>)) {
-  next unless $line =~ /^(\S+)\s+U\+(\S+)(?:\s+U\+(\S+))?/;
-  $ENTITIES{$1} = defined $3 ? (chr(hex $2) . chr(hex $3)) : chr(hex $2);
+{
+  # Don't use Mojo::File here due to circular dependencies
+  my $html_path = File::Spec->catfile(dirname(__FILE__),
+    'resources', 'html_entities.txt');
+
+  open my $html_fh, '<', $html_path
+    or croak 'Unable to open html entities file';
```

Comment:
I don't like hiding the actual error, especially when it would be such an unexpected exception.